datumaro/components/format_detection.py

```diff
@@ -457,9 +465,6 @@ def apply_format_detector(
     """
     context = FormatDetectionContext(dataset_root_path)
 
-    if not osp.isdir(dataset_root_path):
-        context.fail(f"root path {dataset_root_path} must refer to a directory")
```

Comment:
Why do we remove this line?

Reply:
`dataset_root_path` can be a file not just a directory to provide import feature from a single arrow file.

Reply:
`root_path` literally means directory, and entering a directory path has traditionally been the behavior of this function that globally affects to all Datumaro components. It looks like it might be possible to follow this rule in the ArrowDataset (by fixing filename pattern).

Reply:
I changed it in this way because the dataset can be saved into a single arrow file. I felt that since we can export into a single file, we need to also import from a single file as well.

Furthermore, doc indicates it can receive a single file as well.
https://github.com/openvinotoolkit/datumaro/blob/37b62bb5fef85211ebc8f804bf0e20a02e2fb7ad/datumaro/components/dataset.py#L1350

Limiting the input as directory for all formats would not be a preferable solution, I believe.

Reply:
My opinion is that I don't understand why a change that affects the overall behavior is necessary in this PR although there are ways to avoid it. Even defining the data format based on the directory structure is a rule followed by all datasets except `ArrowDataset`.  Therefore, it is recommended to inspect the directory structure rather than probing the data file in `detect()`. Lastly, the only one who consumes `apply_format_detector()` is `detect_dataset_format()` and `detect_dataset_format()` is only consumed by
https://github.com/openvinotoolkit/datumaro/blob/37b62bb5fef85211ebc8f804bf0e20a02e2fb7ad/datumaro/components/environment.py#L252-L277
and L276 shows that it doesn't like a file path.

Reply:
Fair point. Reverted in https://github.com/openvinotoolkit/datumaro/pull/931/commits/44eff3a90ef23a86adeea5ca3a4258b80f88f811